syntax = "proto3";

package meeting;

service MeetingService {
  rpc SaveImportance (SaveImportanceRequest) returns (SaveImportanceResponse);
  rpc TranscribeAndSummarize (TranscribeAndSummarizeRequest) returns (TranscribeAndSummarizeResponse);
  rpc SaveMetadata (SaveMetadataRequest) returns (SaveMetadataResponse);
  rpc EditMetadata (EditMetadataRequest) returns (EditMetadataResponse);
  rpc DeleteMetadata (DeleteMetadataRequest) returns (DeleteMetadataResponse);
}

message SaveImportanceRequest {
  string item_name = 1;
  int32 importance = 2;
  string details = 3;
}

message SaveImportanceResponse {
  bool success = 1;
  string message = 2;
}

message TranscribeAndSummarizeRequest {
  bytes audio_data = 1;
  string audio_format = 2;
}

message TranscribeAndSummarizeResponse {
  string transcription = 1;
  string summary = 2;
  string error_message = 3;
}

message SaveMetadataRequest {
  string title = 1;
  string meeting_date = 2;
  repeated string attendees = 3;
  string additional_info = 4;
}

message SaveMetadataResponse {
  string meeting_id = 1;
  bool success = 2;
  string message = 3;
}

message EditMetadataRequest {
  string meeting_id = 1;
  optional string title = 2;
  optional string meeting_date = 3;
  repeated string attendees = 4;
  optional string additional_info = 5;
}

message EditMetadataResponse {
  bool success = 1;
  string message = 2;
}

message DeleteMetadataRequest {
  string meeting_id = 1;
}

message DeleteMetadataResponse {
  bool success = 1;
  string message = 2;
}
